### PR TITLE
[Apache Tomcat] Migrate to GA

### LINKED ISSUE #7363: [O11y] Make Apache Tomcat GA
- Make Apache Tomcat GA with the following data streams
   - Metrics:
       - access
       - cache
       - catalina
       - connection_pool
       - localhost
       - memory
       - request
       - session
       - thread_pool

### Enhancement Requests
- #7204
   - Support multiline parser for multiple date formats and support Log4j format for the Catalina and Localhost data streams
- #7635
   - The forwarded tag will enable the `publisher_pipeline.disable_host: true`, which is the functionality of a Filebeat input setting that only applies to Filebeat-based integrations. The Apache Tomcat metrics data streams are using Metricbeat-based input. Hence need to remove the forwarded tag for metrics data streams.
- #7634 
   - In the Filebeat-based integrations, support the description related to the forwarded tag information.
- #7584
   - A user is testing to ingest JIRA access logs via Tomcat integration. However, the log parsing fails because of the wrong log pattern. Need to support the log format that is being used.
- #7773
   - If the user hasn't `%A %X %T` or one of them, need to update the grok pattern by making the fields optional so that user can ingest the data without any error.
- #7622
   - Fix ingest pipeline tests and update X-forwared-for IP field from `header_forwarder` to `client.ip`

### PR DESCRIPTION
**Pull request by harnish-elastic**
## What does this PR do?

Migrates the integration from BETA to GA.

## Related issues

- Closes #7363

### CODE REVIEW

**Review by milan-elastic — APPROVED** [review]

**Comment by muthu-mps** [comment]
@harnish-elastic -  Here is my observations,

- As the Dynamic ECS mapping is [enabled](https://github.com/elastic/integrations/blob/main/packages/apache_tomcat/_dev/build/build.yml) by default for this integration. Are we planning to have this feature enabled? When we enable the dynamic ECS we can delete the `ecs.yml` file but this file is present in all the datastreams. Is there a any specific reason to keep the ECS files? 
- The dasdhboard screenshot for access log is not aligned properly. The pie chart representation at the bottom could be aligned better with the above charts.
Screenshot 2023-11-07 at 11 44 14 AM

- The access log definition in the document represents the `overall performance` [here](https://docs.elastic.co/integrations/apache_tomcat#:~:text=information%20related%20to-,overall%20performance%20of%20Java%20applications,-.). We need to change the definition for access log datastream to represent the log data thats processed for access logs.

**Comment by harnish-elastic** [comment]
> @harnish-elastic - Here is my observations,
> 
> * As the Dynamic ECS mapping is [enabled](https://github.com/elastic/integrations/blob/main/packages/apache_tomcat/_dev/build/build.yml) by default for this integration. Are we planning to have this feature enabled? When we enable the dynamic ECS we can delete the `ecs.yml` file but this file is present in all the datastreams. Is there a any specific reason to keep the ECS files?

Please refer to this old [thread](https://github.com/elastic/integrations/pull/5424#discussion_r1185739004) for the discussion on this. We kept the ecs.yml so that the fields would be visible in README.


> * The dasdhboard screenshot for access log is not aligned properly. The pie chart representation at the bottom could be aligned better with the above charts.
> Screenshot 2023-11-07 at 11 44 14 AM
> * The access log definition in the document represents the `overall performance` [here](https://docs.elastic.co/integrations/apache_tomcat#:~:text=information%20related%20to-,overall%20performance%20of%20Java%20applications,-.). We need to change the definition for access log datastream to represent the log data thats processed for access logs.

Sure, let me update it, Thanks!

File: packages/apache_tomcat/_dev/build/docs/README.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ Logs help you keep a record of events that happen on your machine. The `Log` dat
 Metrics give you insight into the statistics of the Apache Tomcat. The `Metric` data streams collected by the Apache Tomcat integration are `cache`, `connection pool`, `memory`, `request`, `session` and `thread pool`, so that the user can monitor and troubleshoot the performance of the Apache Tomcat instance.
 
 Data streams:
-- `access`: Collects information related to overall performance of Java applications.
+- `access`: Collects information related to HTTP transactions between the client and the application server such as IP address of the client sending the request, the request method and URL path, the HTTP status code, the number of bytes returned to the client and more.
```
Comment (muthu-mps) [inline packages/apache_tomcat/_dev/build/docs/README.md:22]:
```suggestion
- `access`: Collects information related to the HTTP transactions, client IP, response code and request processing time.
```

Comment (muthu-mps) [reply packages/apache_tomcat/_dev/build/docs/README.md:22]: Please make the above change [here](https://docs.elastic.co/integrations/apache_tomcat#:~:text=logs%20related%20to-,overall%20performance,-of%20Java%20applications) as well.

Comment (harnish-elastic) [reply packages/apache_tomcat/_dev/build/docs/README.md]: Updated, thanks

**Comment by muthu-mps** [comment]
LGTM!
Adding @ishleenk17 to do a peer review.

**Review by ishleenk17 — APPROVED** [review]
Looks good!

**Pull request merged**